src/chainstate/burn/operations/mod.rs

```diff
@@ -308,6 +313,19 @@ pub struct DelegateStxOp {
     pub burn_header_hash: BurnchainHeaderHash, // hash of the burn chain block header
 }
 
+#[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize)]
+pub struct PegInOp {
+    pub address: StacksAddress,
```

Comment:
I know SIP-021's wire format appendix says that this is a Stacks address, but the text itself says that this can also be a smart contract address.  Can you make this the case?

Bytes 3-24 would be the address portion (1 byte version, 20 byte hash), and bytes 24-64 would be the name portion.

The name would be a smart contract name, so it would be between 5 and 40 bytes long.

Reply:
Sure thing, I'll update.

Reply:
I'll represent the smart contract name as a String for the alpha version, leaving room for a future discussion around improved type safety for contract addresses.

The reason is that I have not found any good existing types to represent smart contract addresses. The best type I know of is `clarity::vm::types::PrincipalData`, which uses `ContractName` to validate contract names. However, this would make `chainstate::burn::operations` depend on the `clarity` crate, which I would not like to do on a whim. Also, `ContractName` can introduce unintentional panics as observed in #3431, which would risk bringing down a running node on a malformed transaction.

I guess ideally it would be best to have type alongside `StacksAddress` in `stacks_common::types::chainstate` to represent contract addresses.

Let me know any thoughts on this.

Reply:
We discussed this on an sBTC working group meeting today. We're okay with depending on clarity types in `stacks_common::types::chainstate`, so I will change this ti use the `PrincipalData` type.

Reply:
Done :heavy_check_mark: